### PR TITLE
[gui] drop video/music overlay dialogs

### PR DESCRIPTION
**Pull request by xhaggi**
As discussed with @phil65 it makes sense to drop video/music overlay dialogs because a skin could easily create a custom dialog which acts as a replacement. Most skins don't use these dialogs but have to include the corresponding skin xml's.

@ronie @BigNoid @MartijnKaijser @Montellese mind taking a look

### CODE REVIEW

**Comment by phil65** [comment]
@ronie would be great if you could check if there is any repo skin at all which uses that dialog. As @xhaggi already pointed out, those dialogs are basically the same as creating a custom modeless dialog with Player.HasAudio / HasVideo as a visible condition.

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -428,8 +428,8 @@ const infomap mediacontainer[] = {{ "hasfiles",         CONTAINER_HASFILES },
                                   { "pluginname",       CONTAINER_PLUGINNAME },
                                   { "viewmode",         CONTAINER_VIEWMODE },
                                   { "totaltime",        CONTAINER_TOTALTIME },
-                                  { "totalwatched",     CONTAINER_TOTALWATCHED },
-                                  { "totalunwatched",   CONTAINER_TOTALUNWATCHED },
+                                  { "totalwatched",     CONTAINER_TOTALWATCHED },
+                                  { "totalunwatched",   CONTAINER_TOTALUNWATCHED },
```
Comment (MartijnKaijser) [inline xbmc/GUIInfoManager.cpp]: Unrelated changes?

**Comment by phil65** [comment]
@HitcherUK, @jurialmunkey, @Tgxcorporation, @gade01 does anyone of you use this?

**Comment by BigNoid** [comment]
@xhaggi Theres one reference still that needs removal:
https://github.com/xbmc/xbmc/blob/master/xbmc/music/dialogs/Makefile#L3

@phil65 repo skins are located [here] (https://github.com/xbmc/repo-skins)

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (xhaggi) [inline xbmc/GUIInfoManager.cpp]: i've separated it in a cosmetic commit

**Comment by xhaggi** [comment]
@BigNoid done

**Comment by HitcherUK** [comment]
> @HitcherUK, @jurialmunkey, @Tgxcorporation, @gade01 does anyone of you use this?

Good to go but a quick question -

These 2 dialogs were always rendered on top of everything else so if we wanted to recreate them using a custom dialog with a high zorder would this still be the case?

Thanks.

**Comment by phil65** [comment]
yes, that should still be the case because we still respect the ```<zorder>``` value.

**Comment by HitcherUK** [comment]
Cheers Phil.

**Comment by ronie** [comment]
> @ronie would be great if you could check if there is any repo skin at all which makes proper use of those dialogs.

they are currently used by these skins:
- back-row
- box
- maximinimalism
- metropolis
- refocus

**Comment by phil65** [comment]
@MartijnKaijser would be great if you could provide test builds.
@jingai @jeroenpardon,@chrisbevan, it would be great if you could test this once test builds are available.
To convert those dialogs to "custom" modeless dialogs you would have to change the filename to "Custom.....xml" and add a new window id, a visible condition "player.hasaudio/hasvideo" and probably a zorder of 2 to those dialogs. ( @xhaggi correct me if i´m wrong ;) )

**Comment by MartijnKaijser** [comment]
http://jenkins.kodi.tv/job/BuildMulti-PR-Manually/142/ 
Test builds are building....

**Comment by xhaggi** [comment]
any updates?

**Comment by phil65** [comment]
I will write some PMs on forums, perhaps they dont really check Github.

**Comment by DjCisco** [comment]
I am very unfamiliar with this. What test builds? Test builds of what? How do I test? And what should I test? The feasibility of implementing the overlay dialogs with custom dialogs?

As I understand the previous comments, it is clear that that is possible. So am I supposed to conform the assumptipns of the previous comments by testing? or should I simply remove the dialogs from the skin?

Regards

Francisco

**Comment by phil65** [comment]
Yes, it would be nice if you could test whether you run into any problems when implementing those 2 dialogs as a custom dialog. 
You dont really need the test build for that when thinkin about it again, just a quick confirmation that we didnt miss anything would be great.

**Comment by DjCisco** [comment]
Ok

**Comment by SharpeUK** [comment]
I'll do likewise.

**Comment by chrisbevan** [comment]
Tested. No problems to report. Thanks!

**Comment by xhaggi** [comment]
any objection if we shove it in?

**Comment by chrisbevan** [comment]
Not from me. Go right ahead.

**Comment by phil65** [comment]
yup, good to go IMO.

**Comment by ronie** [comment]
get it in :-)

**Comment by xhaggi** [comment]
okay, jenkins build this please
